src/runtime/virtcontainers/sandbox.go

```diff
@@ -992,6 +1001,111 @@ func (cw *consoleWatcher) stop() {
 	}
 }
 
+func (s *Sandbox) addSwap(ctx context.Context, swapID string, size int64) (*config.BlockDrive, error) {
+	swapFile := filepath.Join(getSandboxPath(s.id), swapID)
+
+	swapFD, err := os.OpenFile(swapFile, os.O_CREATE, 0600)
+	if err != nil {
+		err = fmt.Errorf("creat swapfile %s fail %s", swapFile, err.Error())
+		s.Logger().WithError(err).Error("addSwap")
+		return nil, err
+	}
+	swapFD.Close()
+	defer func() {
+		if err != nil {
+			os.Remove(swapFile)
+		}
+	}()
+
+	// Check the size
+	pagesize := os.Getpagesize()
+	// mkswap refuses areas smaller than 10 pages.
+	size = int64(math.Max(float64(size), float64(pagesize*10)))
+	// Swapfile need a page to store the metadata
+	size += int64(pagesize)
+
+	err = os.Truncate(swapFile, size)
+	if err != nil {
+		err = fmt.Errorf("truncate swapfile %s fail %s", swapFile, err.Error())
+		s.Logger().WithError(err).Error("addSwap")
+		return nil, err
+	}
+
+	err = exec.CommandContext(ctx, "/sbin/mkswap", swapFile).Run()
```

Comment:
- Might be worth making `/sbin/mkswap` a top-level `const` for maximum visibility.
- Unlike `swapon(8)`, if this fails, there are no useful return values (it's just `0` or `1`). As such, I think we should capture stdout+stderr and log these on error. Something like:
  ```go
  var outbuf, errbuf bytes.Buffer
  cmd := exec.CommandContext(ctx, "/sbin/mkswap", swapFile)
  cmd.Stdout = &outbuf
  cmd.Stderr = &outbuf
  err := cmd.Run()
  ```
  In fact, we should call all external commands using a wrapper function that does this imho.

Reply:
Added.